ml-proto/README.md

```diff
@@ -5,10 +5,13 @@ This repository implements a prototypical reference interpreter for WebAssembly.
 Currently, it can
 
 * *parse* a simple S-expression format,
+* *decode* the binary format (work in progress),
 * *validate* modules defined in it,
-* *execute* invocations to functions exported by a module.
+* *execute* invocations to functions exported by a module,
+* *encode* the binary format,
+* *prettyprint* the S-expression format (work in progress).
 
-The file format is a (very dumb) form of *script* that cannot just define a module, but also batch a sequence of invocations.
+The S-expression format is a (very dumb) form of *script* that cannot just define a module, but in fact a sequence of them, and a batch of invocations, assertions, and conversions to each one. As such it is a superset of the binary format, with the additional functionality purely intended as testing infrastructure. (See [below][#scripts] for details.)
```

Comment:
I'll make one more argument against calling .wast a "superset" of a .wasm:

From a user perspective, as opposed to a ml-proto implementation perspective, .wast files are not modules; they are scripts that have a way to define modules inline. They don't satisfy the "isa" relationship, except via an awkward path where all modules but one are silently discarded, behavior which isn't useful outside of artificial testcases.